src/modules/market/components/market-preview/market-preview.jsx

```diff
@@ -10,31 +10,45 @@ import toggleHeight from "utils/toggle-height/toggle-height";
 
 import CommonStyles from "modules/market/components/common/market-common.styles";
 import Styles from "modules/market/components/market-preview/market-preview.styles";
-import MarketAdditonalDetails from "modules/reporting/components/market-additional-details/market-additional-details";
+import MarketAdditonalDetails from "modules/market/components/market-additional-details/market-additional-details";
 import { isEmpty } from "lodash";
 import ToggleHeightStyles from "utils/toggle-height/toggle-height.styles";
 
 export default class MarketPreview extends Component {
   static propTypes = {
     testid: PropTypes.string,
-    id: PropTypes.string,
+    id: PropTypes.string.isRequired,
     isLogged: PropTypes.bool.isRequired,
-    toggleFavorite: PropTypes.func,
+    toggleFavorite: PropTypes.func.isRequired,
```

Comment:
`toggleFavorite` is undefined when MarketPreview is uses from reporting-disptue.jsx and `collectMarketCreatorFees` is undefined when MarketPreview is used in reporting-report.jsx